src/Commands/ContinuousIntegration/TravisCi/Config/ConfigBuilder.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace QuickStrap\Commands\ContinuousIntegration\TravisCi\Config;
+
+use QuickStrap\Commands\ContinuousIntegration\TravisCi\TravisCiQuestionHelper;
+use QuickStrap\Helpers\Composer\PackageHelper;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class ConfigBuilder
+{
+    /** @var  string[] */
+    private static $allowedPhpVersions = array(
+        '7.0',
+        '5.6',
+        '5.5',
+        '5.4',
+        '5.3',
+        '5.2',
+        'hhvm'
+    );
+
+    /** @var  TravisCiQuestionHelper */
+    private $questionHelper;
+
+    /** @var PackageHelper */
+    private $packageHelper;
+
+    /**
+     * ConfigurationBuilder constructor.
+     * @param TravisCiQuestionHelper $questionHelper
+     * @param PackageHelper $packageHelper
+     */
+    public function __construct(TravisCiQuestionHelper $questionHelper, PackageHelper $packageHelper)
+    {
+        $this->questionHelper = $questionHelper;
+        $this->packageHelper = $packageHelper;
+    }
+
+    /**
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     * @return Config
+     * @throws \Symfony\Component\Console\Exception\InvalidArgumentException
+     * @throws \Symfony\Component\Console\Exception\RuntimeException
+     */
+    public function createConfig(InputInterface $input, OutputInterface $output)
+    {
+        $config = new Config('php');
+
+        // Configure the PHP versions to build.
+        $versions = $this->questionHelper->askWhichPhpVersionsToBuild($input, $output, self::$allowedPhpVersions);
+        if (is_array($versions)) {
+            $config->setPhp($versions);
+
+            // Configure the builds that are allowed to fail for particular PHP versions.
+            $allowedToFail = $this->questionHelper->askWhichBuildsAreAllowedToFail($input, $output, $versions);
+            if (is_array($allowedToFail)) {
+                $config->getMatrix()->setAllowFailures($allowedToFail);
```

Comment:
```
matrix:
  allow_failures:
    - php: 5.5
    - php: 5.6
    - php: 7.0
```
Decodes to this in PHP (via `Yaml::parse`):
```
Array
(
    [matrix] => Array
        (
            [allow_failures] => Array
                (
                    [0] => Array
                        (
                            [php] => 5.5
                        )

                    [1] => Array
                        (
                            [php] => 5.6
                        )

                    [2] => Array
                        (
                            [php] => 7
                        )

                )

        )

)
```

Which then becomes yml (via `Dumper::dump` [with an inline value of 3]):
```
matrix:
    allow_failures:
        - { php: 5.5 }
        - { php: 5.6 }
        - { php: '7.0' }
```

Which decodes to PHP again:
```
Array
(
    [matrix] => Array
        (
            [allow_failures] => Array
                (
                    [0] => Array
                        (
                            [php] => 5.5
                        )

                    [1] => Array
                        (
                            [php] => 5.6
                        )

                    [2] => Array
                        (
                            [php] => 7.0
                        )

                )

        )

)
```


So I think if you use nested arrays it will work out.